Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -437,12 +437,8 @@ extension ChatChannel {
 
         let reads: [ChatChannelRead] = try dto.reads.map { try $0.asModel() }
 
-        let unreadCount: () -> ChannelUnreadCount = {
```

Comment:
we don't need the `isValid` checks anymore?

Reply:
Because it runs immediately then I don't see any reason why we would need it. Made sense when this was called at a random point of time in the future.

Reply:
Yeah in theory this should not be needed anymore

Reply:
Either way, as next step we should try to avoid these DB calls, and use the payloads data directly

Reply:
That would be the next logical step, I also have some ideas on how to do it.

Reply:
For now I kept it as simple as possible and just converted these closures to run immediately.